aether-ui/aether/ui/assets/apps/pipeline/PipelineList.jsx

```diff
@@ -22,19 +22,210 @@ import React, { Component } from 'react'
 import { connect } from 'react-redux'
 import { FormattedMessage } from 'react-intl'
 
-import { ModalError, NavBar } from '../components'
+import { ModalError, NavBar, Modal } from '../components'
 
 import PipelineNew from './components/PipelineNew'
 import PipelineCard from './components/PipelineCard'
+import DeleteStatus from './components/DeleteStatus'
 
-import { getPipelines } from './redux'
+import { getPipelines, deletePipeline, pipelineChanged } from './redux'
 
 class PipelineList extends Component {
   constructor (props) {
     super(props)
 
     // fetch pipelines list
     props.getPipelines()
+
+    this.state = {
+      showDeleteModal: false,
+      showDeleteProgress: false,
+      deleteOptions: {
+        schemas: false,
+        entities: false,
+        submissions: false
+      }
+    }
+
+    this.deletePipeline = this.deletePipeline.bind(this)
+    this.onDelete = this.onDelete.bind(this)
+    this.hideModalProgress = this.hideModalProgress.bind(this)
+  }
+
+  onDelete (pipeline) {
+    this.setState({ showDeleteModal: true })
+    this.props.pipelineChanged(pipeline)
+  }
+
+  deletePipeline () {
+    this.setState({
+      showDeleteModal: false,
+      showDeleteProgress: true
+    })
+    this.props.deletePipeline(
+      this.props.pipeline.id,
+      this.state.deleteOptions
+    )
+  }
+
+  renderDeletionModal () {
```

Comment:
Wouldn't this be the same piece of code included in the `Pipeline` component? :see_no_evil: